Migrate to Rails 5 on Heroku

mattharris5 opened the pull request.

mattharris5 committed:
Setup for heroku deployment

mattharris5 committed:
Only require sqlite3 for development

mattharris5 committed:
Fixing precompile error

mattharris5 committed:
Removing unused tango icon set

mattharris5 committed:
Adding precompiled assets

mattharris5 committed:
Remove smtp config block

mattharris5 committed:
Redirect logging to STDOUT

mattharris5 committed:
Add Amazon RDS cert bundle

mattharris5 committed:
Switch to postgres

mattharris5 committed:
Upgrade rollbar gem

mattharris5 committed:
Temporarily disable sucker_punch for rollbar

mattharris5 committed:
Disable SSL fix to work with heroku

mattharris5 committed:
Switch to use postgres schemas for apartment

mattharris5 committed:
Switching to `pluck`

mattharris5 committed:
Force SSL in production

mattharris5 committed:
Convert `ialias` to allow more than 255 chars

Some of the institution records include aliases that are longer than 255

mitchellharper12 committed:
Begin Rails 4 upgrade

mitchellharper12 committed:
Add Rails modules moved to gems

mattharris5 committed:
Merge branch 'master' into migration/heroku

mattharris5 committed:
Gem and schema cleanup

mattharris5 commented:
Lots of work to do before this can be merged...

mattharris5 committed:
Merge branch 'master' into migration/heroku

mattharris5 committed:
Configure filters to warn on false

Previously all filters resulted in a green check or a red X, but that
makes it hard to identify which items are just informational and which
need action taken. This commit adds the ability to specify which
filters should trigger a warning, and updates the views to accommodate
this change.

mattharris5 committed:
Refactor filter cache to use redis and better speed

* Provides an arbitrary redis object scoped to each customer using its
own namespace. This is accessible with `Customer.redis` throughout the
application.
* Moves filter cache out of the object model and into the redis cache
store.
* Stores the full filter result (e.g., “fail warn”) instead of just
`true/false`
* Populates multiple representations of the filter cache in redis
beyond just each person’s filter results. Also populates a hash of
filter warning counts, as well as sets for each filter+state that
stores the matching ID’s for that filter. For example, the
`ObjectFilter:5:pass` set holds all participant ID’s that pass
ObjectFilter #5.
* Decouples the `update_filter_cache` process from the record saving
process, instead using a background worker to update the cache after
the record is saved.
* Takes advantage of redis’ `SINTER` intersection command to find the
matching set against lots of filter conditions on the participant index
page. Speeds up page load time considerably.
* Adds a mouseover popup on the participant row that shows which
filters are in the “warn” status.

mattharris5 committed:
Heroku rails 4 changes

mattharris5 committed:
Merge branch 'master' into migration/heroku

# Conflicts:
#	Gemfile
#	Gemfile.lock

mattharris5 committed:
Rails 4.2 changes

mattharris5 committed:
Merge branch 'migration/rails4' into migration/heroku

# Conflicts:
#	Gemfile
#	Gemfile.lock

mattharris5 committed:
Routes and other rails4 fixes

mattharris5 committed:
Update to ruby 1.9 hash syntax

Used `find . -name \*.rb -exec perl -p -i -e
's/([^:]):(\w+)\s*=>/\1\2:/g' {} \;` to automatically convert all
old-style hashes.

mattharris5 committed:
Update `default_scope` to rails 4.2 syntax

mattharris5 committed:
Switch to safer regex anchors

mattharris5 committed:
Update `scope` syntax for rails 4.2

mattharris5 committed:
Remove previous monkey-patch for Identity

mattharris5 committed:
Use new SimpleCalendar syntax

mattharris5 committed:
Use safe regex to validate URL

mattharris5 committed:
Use new finder methods

mattharris5 committed:
Merge branch 'feature/filter-warnings-and-caching' into migration/heroku

# Conflicts:
#	Gemfile
#	Gemfile.lock
#	app/controllers/participants_controller.rb
#	app/models/customer.rb
#	app/models/object_filter.rb
#	config/initializers/session_store.rb
#	db/schema.rb

mattharris5 committed:
Temporarily disable or fix some old rails methods

mattharris5 committed:
WIP adding Twitter Bootstrap through sprockets

mattharris5 committed:
Remove uses of `Model.find :all`

mattharris5 committed:
Add turbolinks and jQuery 2

mattharris5 committed:
Adds bootstrap and FontAwesome

mattharris5 committed:
Switch to native bootstrap nav

mattharris5 committed:
Start CSS cleanup and use SASS

mattharris5 committed:
Remove `@layout_in_blocks` style

mattharris5 committed:
Navbar and header

mattharris5 committed:
Move participation info to /my/participation

mattharris5 committed:
CSS and HTML cleanup and reduction

mattharris5 committed:
Switch to use `content_for` for sidebar content

mattharris5 committed:
Convert to ruby 1.9 style hash syntax

mattharris5 committed:
Style, layout, and form improvements

mattharris5 committed:
Use bootstrap tables as the default for all tables

mattharris5 committed:
Primary style overrides

mattharris5 committed:
Do not show sidebar unless content provided

mattharris5 committed:
Add date picker for dates

mattharris5 committed:
Form structure improvements

mattharris5 committed:
Style improvements and subheader

mattharris5 committed:
Update syntax for `find` and `friendly_id`

mattharris5 committed:
Use shorthand for injecting operations subhead links

aayush19saxena committed:
Implemented search using searchkick with multi-tenancy

mattharris5 commented on app/controllers/session_controller.rb:
```diff
@@ -5,9 +5,9 @@ class SessionController < ApplicationController
-    redirect_to locator_url(subdomain: false) if Customer.current_customer.nil? || Customer.current_customer.new_record?
-    redirect_to "/auth/#{Customer.allowable_login_methods_list.first}" if Customer.allowable_login_methods_list.size == 1 && session[:external_login_context] != :rsvp
-    render layout: "basic"
+    redirect_to locator_url(subdomain: false) and return if Customer.current_customer.nil? || Customer.current_customer.new_record?
```
I don't think we need this... Was the login page not working without this? Also be sure to keep things like this separated into a different commit since I don't think this has anything to do with searchkick.

aayush19saxena replied:
Hey Matt, I was getting a double render error at the login page. "Render and/or redirect were called multiple times in this action. Please note that you may only call render OR redirect, and at most one time)" sorry about the commit.

mattharris5 replied:
Interesting. I will have to look at that and figure out why it wasn't showing that for me. If it's double rendering then that means we have some other condition that's not hitting the existing `if` statements, I think.

mattharris5 commented on config/initializers/api_keys.rb:
```diff
@@ -1,4 +1,2 @@
-resp = s3.get_object( bucket: ENV['S3_BUCKET_NAME'], key: ENV['API_KEYS_PATH'] )
-API_KEYS = YAML::load(resp.body.read)
-Rails.logger.info { "Successfully loaded keys from S3" }
+config_file_path = File.join(ENV['SHARED_CONFIG_ROOT'] || "#{Rails.root}/config", "api-keys.yml")
```
Please be very careful about making changes like this!! This is going to be break the app, so I'll revert it.

mattharris5 committed:
Switch to bootstrap tabs

mattharris5 committed:
Remove web console due to security risk

mattharris5 committed:
Expands search to be a universal app-wide search

mattharris5 committed:
Remove CollegeMapper integration

mattharris5 committed:
Update turbolinks progress bar color

mattharris5 committed:
Remove tab badge if count is zero

mattharris5 committed:
Fallback to local `api_keys` if available

mattharris5 committed:
Improved search bar hide/show transition

mattharris5 committed:
Improved avatar handling

mattharris5 committed:
Get unit tests working again

mattharris5 committed:
Use strong parameters

mattharris5 committed:
Miscellaneous compatibility fixes

mattharris5 committed:
Switch to livestamp for auto updating timestamps

mattharris5 committed:
Switch to livestamp in timeago helper

mattharris5 committed:
Use event delegation for select-all

mattharris5 committed:
Move actions from sidebar to subheader

mattharris5 committed:
Move participation details to separate page

WIP

mattharris5 committed:
Remove mentor sidebar

mattharris5 committed:
Add universal CSS indicator/spinner

mattharris5 committed:
Refactor filterables

Some WIP with location hash updating and college stage selectors.

mattharris5 committed:
Switch to bootstrap popovers

Replaces previous custom code that was not nearly as reliable or
accurate. Also switches to event delegation.

mattharris5 committed:
Remove avatar

Temporarily maybe?

mattharris5 committed:
Style improvements

mattharris5 committed:
Refactor old AR `find` methods to new `where` pattern

mattharris5 committed:
Remove MultiTenantProxyable

mattharris5 committed:
Remove CollegeMapper integration

mattharris5 committed:
Remove acts_as_xlsx

mattharris5 committed:
Switch Reports to use ActiveJob and ActionCable

mattharris5 committed:
Restructure view JS into App modules

mattharris5 committed:
View and style updates for new structure

mattharris5 committed:
Upgrade to rails 5

mattharris5 committed:
Switch models to inherit from ApplicationRecord

instead of ActiveRecord::Base as in rails < 5

mattharris5 committed:
Move ActiveRecord customizations to Concerns

`ApplicationRecord` then inherits all of the concerns for all models.

mattharris5 committed:
Use ActiveJob for async processes

mattharris5 committed:
Update participants display to use unobtrusive JS

mattharris5 committed:
Disable global select-all checkbox

mattharris5 committed:
Remove mandrill

mattharris5 committed:
Rails 5 updates

mattharris5 committed:
Modify carrier wave processing for reports

mattharris5 committed:
Disable logentries

mattharris5 committed:
Disable miniprofiler

mattharris5 committed:
Use secret key base from ENV in dev as well

mattharris5 committed:
Visual enhancements

mattharris5 committed:
Only use sqlite3 in dev

mattharris5 committed:
Removing unused image assets

Now prefer FontAwesome and bootstrap for all visual elements

mattharris5 committed:
Force the channels directory to remain

(even if it’s empty)

mattharris5 committed:
Refer to the new name of the bcrypt gem

mattharris5 committed:
Fix JS syntax for older versions

mattharris5 committed:
Adds task for verifying assets compile successfully

mattharris5 committed:
Switch to strong_parameters

mattharris5 committed:
Use strong params

mattharris5 committed:
Remove CollegeMapper integration

mattharris5 committed:
Update to latest ActiveResource

With rails 5 compatibility

mattharris5 committed:
Deployment config

mattharris5 committed:
Fetch redis URL from ENV

mattharris5 committed:
Use erb to dynamically pull the ENV var

mattharris5 committed:
Shift to sidekiq concurrency of 5

mattharris5 committed:
Change redis config size

mattharris5 committed:
Remove concurrency specifics

mattharris5 committed:
Login page style improvements

mattharris5 committed:
Use strong parameters for Customer

mattharris5 committed:
Add sticky table header

mattharris5 committed:
Improve pagination and ajax processing

mattharris5 committed:
Move participant tab content under respective controllers

Previously all tab content on `participant#show` were populated with
partials in the `/views/participants` directory. Now tab content is
populated with the appropriate `index.html.*` file from the
corresponding child resource.

Also introduces a global controller listener that watches for presence
of the `X-DreamSIS-Render-Intent` header, which can be set to
`no-layout` to automatically prevent the app layout being rendered in
the response. This is useful for the participant tabs because the
client can request the tab content using the same controller method.

mattharris5 committed:
Auto-load popovers and tooltips

and improve styling

mattharris5 committed:
Cleanup code

mattharris5 committed:
Style improvements

mattharris5 committed:
Move `after_*` hooks to be after commit

Therefore the change only gets logged if the change was committed to
the database.

mattharris5 committed:
Adds acts_as_taggable

mattharris5 committed:
Better structure for badge controls on participant detail

mattharris5 committed:
Better structure for user profile page

mattharris5 committed:
Store student ID number as string

Previously stored as integer, but some student numbers are out of
bounds or include other chars.

mattharris5 committed:
Only use rack-timeout in production

mattharris5 committed:
Use UUIDs for currentRequest tracking

mattharris5 committed:
Ignore arrow keys in some situations

- when modifier keys are also pressed, ignore
- when there are no tabs on the screen, ignore

mattharris5 committed:
Improve mentor views and use delayed loading

mattharris5 committed:
Eliminate shared/proxy events

mattharris5 committed:
Update event index view to new style
